### PR TITLE
Numeric planning

### PR DESCRIPTION
**Pull request by Shi-Raida**
Refactoring and fixing some bugs on resource constraints encoding.

### CODE REVIEW

**Review by arbimo — CHANGES_REQUESTED** [review]
This includes the code for PR #104 that is not ready yet (awaiting the corresponding PR to be merged in UP)

**Review by arbimo — CHANGES_REQUESTED** [review]
This looks good to me. I have just pointed out a few very minor changes and I think it can be merge directly when this is fixed :)

File: solver/src/reif/mod.rs [hunk 1]
```diff
@@ -26,6 +26,28 @@ pub enum ReifExpr {
...
 }
 
+impl std::fmt::Display for ReifExpr {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ReifExpr::Lit(l) => {
+                write!(f, "{l:?}")
+            }
+            ReifExpr::MaxDiff(md) => {
+                write!(f, "{md:?}")
```
[the remark this answers is not in the document]
Comment (Shi-Raida) [reply solver/src/reif/mod.rs:36]: The `Debug` implementation of `DifferenceExpression` (`md`) writes `b <= a + ub`.
Isn't it enough?

[the remark this answers is not in the document]
Comment (arbimo) [reply solver/src/reif/mod.rs:36]: Yes, I misread the `write`.

**Comment by arbimo** [comment]
LGTM. Could you merge the latest master in this branch?
Changes in master related to testing should take precedence over the ones in this PR.

**Review by arbimo — APPROVED** [review]

**Comment by Shi-Raida** [comment]
> > Four problems crash on Anytime but I don't have logs from Aries so it doesn't start
> 
> Most likely unrelated to this PR, I suspect that the code for checking the correctness of anytime planners does not support scheduling problems.

Yes, I find in `planning/unified/plugin/up_aries/solver.py` line 340:

``` python
assert isinstance(problem, up.model.Problem)
```

The four problems are `SchedulingProblem` which does not inherit from `Problem`.
I'll open a new branch to fix it.

**Pull request merged**